pkgs/desktops/plasma-5/plasma-desktop/default.nix

```diff
@@ -42,12 +43,10 @@ mkDerivation {
   CXXFLAGS = [
     "-I${lib.getDev xorgserver}/include/xorg"
     "-I${lib.getDev xf86inputsynaptics}/include/xorg"
+    "-I${lib.getDev xf86inputevdev}/include/xorg"
+    "-I${lib.getDev xf86inputlibinput}/include/xorg"
```

Comment:
I don't really like these `-I` flags.
There are some packaging issues on xf86* which generate .pc files with wrong include dir. Maybe I'll create another PR later.

Reply:
```
result-dev
├── include
│   └── xorg
│       └── libinput-properties.h
├── lib
│   └── pkgconfig
│       └── xorg-libinput.pc
└── nix-support
    └── propagated-build-inputs
```

Reply:
If you check `xorg-libinput.pc`. I'll know it's including `xorg-server`'s include dir, instead of itself. So you need to `-I` its include dir in every downstream packages.

I know `-I` is required here. I just mention another issue.

Reply:
Opened #138029 . Note that that PR is independent from this one.

Reply:
i checked out your pr and applied this patch
```
diff --git a/pkgs/desktops/plasma-5/plasma-desktop/default.nix b/pkgs/desktops/plasma-5/plasma-desktop/default.nix
index 37134f240dc..60dfa987a46 100644
--- a/pkgs/desktops/plasma-5/plasma-desktop/default.nix
+++ b/pkgs/desktops/plasma-5/plasma-desktop/default.nix
@@ -21,7 +21,8 @@ mkDerivation {
   nativeBuildInputs = [ extra-cmake-modules kdoctools ];
   buildInputs = [
     boost fontconfig ibus libcanberra_kde libpulseaudio libXcursor libXft xorgserver
-    libxkbfile phonon xf86inputevdev xf86inputsynaptics xinput xkeyboard_config
+    libxkbfile phonon xinput xkeyboard_config
+    xf86inputlibinput xf86inputevdev xf86inputsynaptics
 
     accounts-qt qtdeclarative qtquickcontrols qtquickcontrols2 qtsvg qtx11extras
 
```

but during build i get 
```
plasma-desktop> /build/plasma-desktop-5.22.4/kcms/mouse/backends/x11/x11_evdev_backend.cpp:26:10: fatal error: evdev-properties.h: No such file or directory
plasma-desktop>    26 | #include <evdev-properties.h>
plasma-desktop>       |          ^~~~~~~~~~~~~~~~~~~~
```

also tried `xf86inputevdev.dev` but same error

Reply:
plasma-desktop seems to forget including evdev. We need a patch. I also upstream it: https://invent.kde.org/plasma/plasma-desktop/-/merge_requests/563

```diff
--- a/kcms/mouse/backends/x11.cmake
+++ b/kcms/mouse/backends/x11.cmake
@@ -15,6 +15,7 @@ set(backend_LIBS
     KF5::WindowSystem
     Qt::X11Extras
     PkgConfig::XORGLIBINPUT
+    PkgConfig::EVDEV
     X11::X11
     X11::Xi
     X11::Xcursor
```

Reply:
Nice! should this pr or your pr be merged first? i think yours should be merged first